### PR TITLE
[BAEL-7501] Immutable and Mutable Object Examples with Tests

### PR DESCRIPTION
**Pull request by Imranalam28**
This pull request introduces comprehensive examples and test cases illustrating the concepts of mutable and immutable objects in Java. The code includes implementations of both types along with test files to validate their behavior.

### Changes:
1. Added `ImmutablePerson.java` representing an immutable object with a test file `ImmutablePersonUnitTest.java`.
2. Introduced `MutableObjectExamplesUnitTest.java`.
3. Included `ImmutableObjectExamplesUnitTest.java` to validate the immutability of `String` and `Integer` with assert statements.

These additions aim to assist developers in understanding the distinctions between mutable and immutable objects and how to handle them in Java.

### CODE REVIEW

File: core-java-modules/core-java-lang-oop-patterns/src/test/java/com/baeldung/objectmutability/ImmutableObjectExamplesUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotSame;
+
+public class ImmutableObjectExamplesUnitTest {
+
+    @Test
+    public void givenImmutableString_whenConcat_thenNotSameAndCorrectValues() {
```
Comment (vinipx) [inline core-java-modules/core-java-lang-oop-patterns/src/test/java/com/baeldung/objectmutability/ImmutableObjectExamplesUnitTest.java:11]: maybe _whenConcatString_ or _whenConcatWord_ ?

File: core-java-modules/core-java-lang-oop-patterns/src/test/java/com/baeldung/objectmutability/ImmutableObjectExamplesUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        String modifiedString = originalString.concat(" World");
+
+        assertNotSame(originalString, modifiedString);
+
+        assertEquals("Hello", originalString);
+        assertEquals("Hello World", modifiedString);
+    }
+
+    @Test
+    public void givenImmutableInteger_whenAdd_thenNotSameAndCorrectValue() {
```
Comment (vinipx) [inline core-java-modules/core-java-lang-oop-patterns/src/test/java/com/baeldung/objectmutability/ImmutableObjectExamplesUnitTest.java:22]: maybe _whenAddInteger_ ?

File: core-java-modules/core-java-lang-oop-patterns/src/test/java/com/baeldung/objectmutability/ImmutablePersonUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class ImmutablePersonUnitTest {
+
+    @Test
+    public void givenImmutablePerson_whenModifyName_thenCompilationError() {
+        ImmutablePerson person = new ImmutablePerson("John", 30);
+        // person.setName("Jane");
```
Comment (vinipx) [inline core-java-modules/core-java-lang-oop-patterns/src/test/java/com/baeldung/objectmutability/ImmutablePersonUnitTest.java:12]: let's remove it

File: core-java-modules/core-java-lang-oop-patterns/src/test/java/com/baeldung/objectmutability/MutableObjectExamplesUnitTest.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class MutableObjectExamplesUnitTest {
+
+    @Test
+    public void givenMutableString_whenModify_thenCorrectValue() {
```
Comment (vinipx) [inline core-java-modules/core-java-lang-oop-patterns/src/test/java/com/baeldung/objectmutability/MutableObjectExamplesUnitTest.java:13]: _whenModify_ or _whenAppendElement_?

**Review by vinipx — CHANGES_REQUESTED** [review]

File: core-java-modules/core-java-lang-oop-patterns/src/test/java/com/baeldung/objectmutability/ImmutableObjectExamplesUnitTest.java (hunk 1 above)
Comment (Imranalam28) [reply core-java-modules/core-java-lang-oop-patterns/src/test/java/com/baeldung/objectmutability/ImmutableObjectExamplesUnitTest.java]: Done

File: core-java-modules/core-java-lang-oop-patterns/src/test/java/com/baeldung/objectmutability/ImmutableObjectExamplesUnitTest.java (hunk 2 above)
Comment (Imranalam28) [reply core-java-modules/core-java-lang-oop-patterns/src/test/java/com/baeldung/objectmutability/ImmutableObjectExamplesUnitTest.java]: Done

File: core-java-modules/core-java-lang-oop-patterns/src/test/java/com/baeldung/objectmutability/ImmutablePersonUnitTest.java (hunk 3 above)
Comment (Imranalam28) [reply core-java-modules/core-java-lang-oop-patterns/src/test/java/com/baeldung/objectmutability/ImmutablePersonUnitTest.java]: Done

File: core-java-modules/core-java-lang-oop-patterns/src/test/java/com/baeldung/objectmutability/MutableObjectExamplesUnitTest.java (hunk 4 above)
Comment (Imranalam28) [reply core-java-modules/core-java-lang-oop-patterns/src/test/java/com/baeldung/objectmutability/MutableObjectExamplesUnitTest.java]: Done

File: core-java-modules/core-java-lang-oop-patterns/src/test/java/com/baeldung/objectmutability/ImmutablePersonUnitTest.java [hunk 5]
```diff
@@ -0,0 +1,18 @@
+package com.baeldung.objectmutability;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class ImmutablePersonUnitTest {
+
+
```
Comment (vinipx) [inline core-java-modules/core-java-lang-oop-patterns/src/test/java/com/baeldung/objectmutability/ImmutablePersonUnitTest.java:9]: there is an extra line here - let's remove it

**Review by vinipx — CHANGES_REQUESTED** [review]
one minor comment left

**Review by vinipx — APPROVED** [review]

**Pull request merged**